src/gt4py/backend/gtc_backend/defir_to_gtir.py

```diff
@@ -129,33 +152,53 @@ def visit_ArgumentInfo(
         return all_params[node.name]
 
     def visit_ComputationBlock(self, node: ComputationBlock) -> List[gtir.VerticalLoop]:
-        stmts = [s for s in self.visit(node.body)]
+        stmts = []
+        temporaries = []
+        for s in node.body.stmts:  # self.visit(node.body):
+            if isinstance(s, FieldDecl) or isinstance(s, VarDecl):
+                dtype = common.DataType(int(s.data_type.value))
+                if dtype == common.DataType.DEFAULT:
+                    dtype = common.DataType.FLOAT64  # TODO properly fix
```

Comment:
@DropD Do you know how to resolve this properly?

Reply:
I am not sure. I definitely think this transformation should not be buried in a visitor method. I would prefer a more centrally defined mapping of DefIR DataType to common.DataType which includes `DEFAULT` -> `FLOAT64`.

Reply:
I think in the frontend you can actually set this default. @egparedes any thoughts?

Reply:
Although we agreed about this feature in the syntax workshop I think it is not yet possible to set a type in the frontend. Anyway, if you don't accept non-explicit types in the new GTIR, I agree with Rico that it would be better to make this DataType conversion more visible. In the long term, this conversion should be gone anyway since the new frontend will get rid of the DefinitionIR